tests/testthat/test-emmeans.R

```diff
@@ -148,3 +148,29 @@ test_that("emmeans works as expected also for weighted model", {
   )
   expect_equal(result_emmeans_df, expected_emmeans_df, tolerance = 1e-3)
 })
+
+test_that("emmeans gives d.f. close to what is expected for weighted model", {
+  skip_if_not_installed("emmeans", minimum_version = "1.6")
+
+  formula <- FEV1 ~ RACE + SEX + ARMCD * AVISIT + us(AVISIT | USUBJID)
+  fev_data <- within(fev_data, AVISIT <- relevel(AVISIT, ref = "VIS4"))
+  fit <- mmrm(formula, fev_data, weights = fev_data$WEIGHT)
+  result <- expect_silent(emmeans::emmeans(fit, pairwise ~ ARMCD*AVISIT))
+  expect_class(result, "emm_list")
+  result_emmeans_df <- as.data.frame(result$emmeans)[, c("ARMCD", "AVISIT", "df")]
+  result_contrasts_df <- as.data.frame(result$contrasts)[, c("contrast", "df")]
+
+  # See design/SAS/R_weighted_nlme.txt for the source of numbers.
+  expected_emmeans_df <- data.frame(
+    ARMCD = factor(c("PBO", "TRT", "PBO", "TRT", "PBO", "TRT", "PBO", "TRT")),
+    AVISIT = factor(c("VIS4", "VIS4", "VIS1", "VIS1", "VIS2", "VIS2", "VIS3", "VIS3"), levels = c("VIS4", "VIS1", "VIS2", "VIS3")),
+    df = c(134, 132, 147, 142, 144, 143, 128, 128)
+  )
+  expect_equal(result_emmeans_df, expected_emmeans_df, tolerance = 1e-2)
+
+  expected_contrasts_df <- data.frame(
+    contrast = c("PBO VIS4 - TRT VIS4", "PBO VIS4 - PBO VIS1", "PBO VIS4 - TRT VIS1", "PBO VIS4 - PBO VIS2", "PBO VIS4 - TRT VIS2", "PBO VIS4 - PBO VIS3", "PBO VIS4 - TRT VIS3", "TRT VIS4 - PBO VIS1", "TRT VIS4 - TRT VIS1", "TRT VIS4 - PBO VIS2", "TRT VIS4 - TRT VIS2", "TRT VIS4 - PBO VIS3", "TRT VIS4 - TRT VIS3", "PBO VIS1 - TRT VIS1", "PBO VIS1 - PBO VIS2", "PBO VIS1 - TRT VIS2", "PBO VIS1 - PBO VIS3", "PBO VIS1 - TRT VIS3", "TRT VIS1 - PBO VIS2", "TRT VIS1 - TRT VIS2", "TRT VIS1 - PBO VIS3", "TRT VIS1 - TRT VIS3", "PBO VIS2 - TRT VIS2", "PBO VIS2 - PBO VIS3", "PBO VIS2 - TRT VIS3", "TRT VIS2 - PBO VIS3", "TRT VIS2 - TRT VIS3", "PBO VIS3 - TRT VIS3"),
```

Comment:
same here. suggest to use the `styler` package with addin to style the selected code, that saves you manual work